### PR TITLE
[Do-not-review] Tests for has-passed-steps with mode steps

### PR DESCRIPTION
**Pull request by NihalHarish**
### Description of changes:
-Added two tests that test the correctness of has_passed_step function with mode steps 

#### Style and formatting:

I have run `pre-commit install` to ensure that auto-formatting happens with every commit.

#### Issue number, if available
#138 
By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: tests/analysis/trials/test_has_passed_step_scenarios.py [hunk 1]
```diff
@@ -38,6 +76,106 @@ def test_single_writer_all_steps_written_complete_job():
...
+    assert num_workers == 1
+    assert trial.loaded_all_steps is False
+    all_steps = trial.steps(show_incomplete_steps=True)
+    completed_steps = trial.steps()
+    assert all_steps == [0, 10, 20, 30, 40, 50, 60, 70]
+    assert completed_steps == all_steps
+    assert trial.has_passed_step(30) == StepState.AVAILABLE
+    assert trial.has_passed_step(23, mode=ModeKeys.TRAIN) == StepState.UNAVAILABLE
+    assert trial.has_passed_step(30, mode=ModeKeys.EVAL) == StepState.AVAILABLE
+    assert trial.has_passed_step(23, mode=ModeKeys.EVAL) == StepState.UNAVAILABLE
```
Comment (Vikas-kum) [inline tests/analysis/trials/test_has_passed_step_scenarios.py]: can you do has_passed_step(40, modeKeys.Train)

File: tests/analysis/trials/test_has_passed_step_scenarios.py [hunk 2]
```diff
@@ -38,6 +76,106 @@ def test_single_writer_all_steps_written_complete_job():
...
+    assert num_workers == 1
+    assert trial.loaded_all_steps is True
+    all_steps = trial.steps(show_incomplete_steps=True)
+    completed_steps = trial.steps()
+    assert all_steps == [0, 10, 20, 30, 40, 50, 60, 70]
+    assert completed_steps == all_steps
+    assert trial.has_passed_step(30) == StepState.AVAILABLE
+    assert trial.has_passed_step(23, mode=ModeKeys.TRAIN) == StepState.UNAVAILABLE
+    assert trial.has_passed_step(30, mode=ModeKeys.EVAL) == StepState.AVAILABLE
+    assert trial.has_passed_step(23, mode=ModeKeys.EVAL) == StepState.UNAVAILABLE
```
Comment (Vikas-kum) [inline tests/analysis/trials/test_has_passed_step_scenarios.py]: please add one more, 
has_passsed_Step(40, modeKeys.TRAIN)

File: tests/analysis/trials/test_has_passed_step_scenarios.py [hunk 3]
```diff
@@ -38,6 +76,110 @@ def test_single_writer_all_steps_written_complete_job():
...
+            trial_dir=path, global_step=i + 40, mode="EVAL", mode_step=i, worker_name="worker_0"
+        )
+
+    trial = create_trial(path)
+    num_workers = len(trial.workers())
+    assert num_workers == 1
+    assert trial.loaded_all_steps is True
+    all_steps = trial.steps(show_incomplete_steps=True)
+    completed_steps = trial.steps()
+    assert all_steps == [0, 10, 20, 30, 40, 50, 60, 70]
```
Comment (Vikas-kum) [inline tests/analysis/trials/test_has_passed_step_scenarios.py]: move this to end of test or another test, There should be unit test only to check has_passed_step , 
create data, create trail, assert has_passed_step

File: smdebug/trials/trial.py [hunk 4]
```diff
@@ -517,11 +517,13 @@ def has_passed_step(self, step, mode=ModeKeys.GLOBAL) -> StepState:
         """
         all_steps = self.steps(mode=mode, show_incomplete_steps=True)
         bisect_idx = bisect_left(all_steps, step)
-        g_step = self._global_step_currently(mode, step)
 
         if bisect_idx < len(all_steps):
+            # This returns either the global step corresponding to the mode-step
+            # or the closest global step that is greater than the step passed as a parameter
+            g_step = self._global_step_currently(mode, all_steps[bisect_left(all_steps, step)])
```
[the remark this answers is not in the document]
Comment (NihalHarish) [reply smdebug/trials/trial.py]: Removed the duplication.

**Pull request merged**